core/src/main/java/org/jruby/ast/ListNode.java

```diff
@@ -174,11 +246,16 @@ public boolean isEmpty() {
     }
     
     public Node get(int idx) {
+        Node single = this.single;
+
+        if (idx == 0 && single != null) return single;
+
         return list[idx];
     }
 
     @Override
     public Iterator<Node> iterator() {
+        Thread.dumpStack();
```

Comment:
:scissors:

Reply:
Fixed! We might want to consider removing this since it doesn't appear to be used.